pkg/option/config.go

```diff
@@ -2322,6 +2322,14 @@ func (c *DaemonConfig) K8sLeasesFallbackDiscoveryEnabled() bool {
 	return c.K8sEnableAPIDiscovery
 }
 
+// DirectRoutingDeviceRequired return whether the Direct Routing Device is needed under
+// the current configuration.
+func (c *DaemonConfig) DirectRoutingDeviceRequired() bool {
+	// BPF NodePort and BPF Host Routing are using the direct routing device now.
+	// When tunneling is enabled, node-to-node redirection will be done by tunneling.
```

Comment:
small nit for readability, feel free to ignore, just makes the code match your comment semantics. 
```suggestion
// When tunneling is enabled, node-to-node redirection will be done by tunneling.
BPFHostRoutingEnabled = !c.EnableHostLegacyRouting
return (c.EnableNodePort || BPFHostRoutingEnabled) && !c.TunnelingEnabled()
```